go/saltpack/errors.go

```diff
@@ -22,6 +22,15 @@ var (
 	// end of the encryption stream.
 	ErrTrailingGarbage = errors.New("trailing garbage found at end of message")
 
+	// ErrFailedToDecodeHeaderLength indicates that we hit an error in the very
```

Comment:
can you double-check that after this refactor you can't delete some dead errors that are never used?